numenc/encoder_decoder.c

```diff
@@ -0,0 +1,947 @@
+        from_int8,
+        METH_VARARGS,
+        "convert an 8-bit signed integer to sortable bytes"
+    },
+    {
+        "to_int8",
+        to_int8,
+        METH_VARARGS,
+        "convert bytes back to a signed 8-bit integer"
+    },
+    {
+        "from_uint8",
+        from_uint8,
+        METH_VARARGS,
+        "convert an 8-bit unsigned integer to sortable bytes"
+    },
+    {
+        "to_uint8",
+        to_uint8,
+        METH_VARARGS,
+        "convert bytes back to an unsigned 8-bit integer"
+    },
+
+    {
+        "from_int16",
+        from_int16,
+        METH_VARARGS,
+        "convert a 16-bit signed integer to sortable bytes"
+    },
+    {
+        "to_int16",
+        to_int16,
+        METH_VARARGS,
+        "convert bytes back to a signed 16-bit integer"
+    },
+    {
+        "from_uint16",
+        from_uint16,
+        METH_VARARGS,
+        "convert a 16-bit unsigned integer to sortable bytes"
+    },
+    {
+        "to_uint16",
+        to_uint16,
+        METH_VARARGS,
+        "convert bytes back to an unsigned 16-bit integer"
+    },
+
+    {
+        "from_int32",
+        from_int32,
+        METH_VARARGS,
+        "convert a 32-bit signed integer to sortable bytes"
+    },
+    {
+        "to_int32",
+        to_int32,
+        METH_VARARGS,
+        "convert bytes back to a signed 32-bit integer"
+    },
+    {
+        "from_uint32",
+        from_uint32,
+        METH_VARARGS,
+        "convert a 32-bit unsigned integer to sortable bytes"
+    },
+    {
+        "to_uint32",
+        to_uint32,
+        METH_VARARGS,
+        "convert bytes back to an unsigned 32-bit integer"
+    },
+
+    {
+        "from_int64",
+        from_int64,
+        METH_VARARGS,
+        "convert a signed 64-bit integer to sortable bytes"
+    },
+    {
+        "to_int64",
+        to_int64,
+        METH_VARARGS,
+        "convert bytes back to a signed 64-bit integer"
+    },
+    {
+        "from_uint64",
+        from_uint64,
+        METH_VARARGS,
+        "convert an unsigned 64-bit integer to sortable bytes"
+    },
+    {
+        "to_uint64",
+        to_uint64,
+        METH_VARARGS,
+        "convert bytes back to an unsigned 64-bit integer"
+    },
+
+    {
+        "from_float32",
+        from_float32,
+        METH_VARARGS,
+        "convert a 32-bit float to sortable bytes"
+    },
+    {
+        "to_float32",
+        to_float32,
+        METH_VARARGS,
+        "convert bytes back to a 32-bit float"
+    },
+
+    {
+        "from_float64",
+        from_float64,
+        METH_VARARGS,
+        "convert a 64-bit float to sortable bytes"
+    },
+    {
+        "to_float64",
+        to_float64,
+        METH_VARARGS,
+        "convert bytes back to a 64-bit float"
+    },
+    {
+        NULL,
+        NULL,
+        0,
+        NULL
+    }
+};
+
+static struct PyModuleDef cModPyDem = {
+    PyModuleDef_HEAD_INIT,
+    "numenc_module",
+    "fast number encoding and decoding in C",
```

Comment:
See PEP 257; the docstring needs to start with a capital letter and an imperative mood.

"Encode and decode numbers to sortable bytes in C"